Tickr/Tickr/SearchTableViewController.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  SearchTableViewController.swift
+//  Tickr
+//
+//  Created by Stephen Payne on 4/10/16.
+//  Copyright © 2016 Stephen Payne. All rights reserved.
+//
+
+import UIKit
+
+class SearchTableViewController: UIViewController {
+    //Class Variables
+    var searchResults: [StockSearchResult] = []
+    
+    //Outlets
+    @IBOutlet weak var tableView: UITableView!
+    @IBOutlet weak var searchBar: UISearchBar!
+    
+    //MARK: - View Lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        searchBar.delegate = self
+    }
+    
+    override func viewDidAppear(animated: Bool) {
+        super.viewDidAppear(animated)
+        
+        searchBar.becomeFirstResponder()
+    }
+
+    /*
+        Will take a search term and query a separate API for possible stocks
+        that match that search term.
+     
+        - parameter searchText: The user input used for finding stocks
+    */
+    func searchYahooFinanceWithString(searchText: String) {
+        
+        //Search for stocks using the user-entered text
+        StockManager.fetchStocksFromSearchTerm(term: searchText) { (stockInfoArray) -> () in
+            dispatch_async(dispatch_get_main_queue(), {
+                //Update the tableView with the search results
+                self.searchResults = stockInfoArray
```

Comment:
Look at the section on "Defining a Capture List" https://developer.apple.com/library/ios/documentation/Swift/Conceptual/Swift_Programming_Language/AutomaticReferenceCounting.html#//apple_ref/doc/uid/TP40014097-CH20-ID52

You need a weak reference to self.searchResults and self.tableView because this creates a retain cycle. Also, if this view controller is popped, it will not release until method completes, so its another memory leak.